cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1066,6 +1066,106 @@ VersionedItem LocalVersionedEngine::append_internal(
     }
 }
 
+VersionedItem LocalVersionedEngine::refresh_symbol_stats_cache() {
+    struct SymbolData {
+        StreamId stream_id_;
+        timestamp first_index_;
+        timestamp last_index_;
+        size_t num_rows_;
+    };
+
+    auto stream_ids = symbol_list().get_symbols(store());
+    std::vector<VersionQuery> version_queries;
+    version_queries.resize(stream_ids.size());
+    auto versions = batch_get_versions(store(), version_map(), stream_ids, version_queries);
+    std::vector<folly::Future<SymbolData>> output;
+    for (auto &&version : folly::enumerate(versions)) {
+        output.emplace_back(std::move(*version).thenValue([store = store()](auto maybe_index_key) -> ReadKeyOutput {
+            util::check(static_cast<bool>(maybe_index_key), "Version not found for stream");
```

Comment:
I think in that edge case I would prefer to fail. What I want to do is to constrain the state space, so that we either move from a valid prior state to a valid end state, or remain obviously in the valid (if out-of-date) prior state. Introducing "resilience" in this sort of operation in my experience produces a sort of "grey failure", where specific components such as symbols repeatedly fail to update, but the problem is not evident because the majority are updated correctly.

Rather than adding premature resilience, sometimes it is better to assume that the real-world likelihood of this actually occurring is very low, and to intentionally design with a slightly brittle adherence to correctness. If it turns out that there are failure conditions we need to tolerate, it will be simple to do so at a later point.

I think I will add the stream id as it will aid in debugging. It does require significant extra lambda-capture though, which is why I avoided it in the first place.